xdsl/dialects/scf.py

```diff
@@ -189,44 +189,61 @@ def get(
         return For(lb, ub, step, iter_args, body)
 
     def verify_(self):
-        if (len(self.iter_args) + 1) != len(self.body.block.args):
+        # op region verification
+        for i, opnd in enumerate(self.operands[:3]):
+            if not isinstance(opnd.type, IndexType):
```

Comment:
This check is not triggered since this is caught by IRDL (hence the drop in coverage for this PR), but it makes sense in the context of #1727 and my question there regarding `signlessIntegerLike`. I'll probably remove this when that question is answered.